ocrd_network/ocrd_network/process_helpers.py

```diff
@@ -14,12 +15,23 @@ def invoke_processor(
         output_file_grps: List[str],
         page_id: str,
         parameters: dict,
+        mets_server_url: Optional[str] = None
 ) -> None:
     if not (processor_class or executable):
         raise ValueError(f'Missing processor class and executable')
     input_file_grps_str = ','.join(input_file_grps)
     output_file_grps_str = ','.join(output_file_grps)
-    workspace = Resolver().workspace_from_url(abs_path_to_mets)
+
+    if mets_server_url:
+        if not mets_server_url.startswith('http://'):
+            if not Path.exists(Path(mets_server_url)):
+                raise RuntimeError(f'Socket file does not exist: {mets_server_url}')
+
+    workspace = Resolver().workspace_from_url(
+        mets_url=abs_path_to_mets,
+        mets_server_url=mets_server_url
+    )
```

Comment:
This would not work for the hostname:port server. So if checking is necessary, why not create the workspace and try to request `/workspace_path`?

Reply:
Yes, I will implement a method that checks that and use it in both the processing server and processing worker.